src/js/providers/video-actions-mixin.js

```diff
@@ -21,20 +22,24 @@ const VideoActionsMixin = {
         if (!width || !height || !this.video.videoWidth || !this.video.videoHeight) {
             return false;
         }
+        const _videotag = this.video;
+        const styles = {
+            objectFit: '',
+            width: '',
+            height: '',
+        };
         if (stretching === 'uniform') {
             // snap video to edges when the difference in aspect ratio is less than 9%
             var playerAspectRatio = width / height;
-            var videoAspectRatio = this.video.videoWidth / this.video.videoHeight;
-            var objectFit = null;
+            var videoAspectRatio = _videotag.videoWidth / _videotag.videoHeight;
             if (Math.abs(playerAspectRatio - videoAspectRatio) < 0.09) {
-                objectFit = 'fill';
+                styles.objectFit = 'fill';
             }
-            style(this.video, {
-                objectFit,
-                width: null,
-                height: null
-            });
         }
+        if (fitVideoUsingTransforms) {
+            fitToBounds(_videotag, width, height, stretching, styles);  
+        } 
+        style(_videotag, styles);
```

Comment:
@boyntoni @johnBartos We should never apply "objectFit" styles, and a hack that attempts to mimic "objectFit" at the same time.

It should always be one of the other. If some versions or Edge or native Android browser actually render "objectFit" properly - then you add the hack on top, it would become over stretched.